### PR TITLE
fix tests except CustomApplicationError if path

### PR DESCRIPTION
**Pull request by jarosik10**

### CODE REVIEW

File: src/app/content/highlights/hooks/createHighlight.spec.ts [hunk 1]
```diff
@@ -76,18 +76,53 @@ describe('createHighlight', () => {
       .mockRejectedValue(error);
     const mock = createMockHighlight();
 
+    try {
     await hook(createHighlight(mock, meta));
+    } catch (error) {
+      expect(createHighlightClient).toHaveBeenCalledWith({highlight: mock});
```
Comment (PiotrKozlowski) [inline src/app/content/highlights/hooks/createHighlight.spec.ts:82]: there is no `expect` about `error`

File: src/app/content/highlights/hooks/locationChange.spec.ts [hunk 2]
```diff
@@ -149,40 +149,63 @@ describe('locationChange', () => {
   });
 
   describe('error handling', () => {
-    it('doesn\'t show a toast if hook ran because of page\'s focus changing', async() => {
+    it('throws HighlightLoadError', async() => {
```
Comment (PiotrKozlowski) [inline src/app/content/highlights/hooks/locationChange.spec.ts:152]: the old test should not be removed

Comment (PiotrKozlowski) [reply src/app/content/highlights/hooks/locationChange.spec.ts:152]: I see that there some checks for receivePageFocus, but I have to look if they are correct

File: src/app/content/highlights/hooks/printHighlights.spec.ts [hunk 3]
```diff
@@ -60,21 +59,16 @@ describe('printHighlights', () => {
       getState: store.getState,
     };
 
-    addPromise = jest.spyOn(helpers.promiseCollector, 'add');
     dispatch = jest.spyOn(helpers, 'dispatch');
+    calmSpy = jest.spyOn(helpers.promiseCollector, 'calm')
+    .mockImplementation(() => Promise.resolve());
```
Comment (PiotrKozlowski) [inline src/app/content/highlights/hooks/printHighlights.spec.ts:64]: indentation

File: src/app/content/highlights/hooks/receiveDeleteHighlight.spec.ts [hunk 4]
```diff
@@ -63,17 +61,41 @@ describe('receiveDeleteHighlight', () => {
     const deleteHighlightClient = jest.spyOn(helpers.highlightClient, 'deleteHighlight')
       .mockRejectedValue(error);
 
-    hook(receiveDeleteHighlight(highlight as unknown as Highlight, meta));
-    await Promise.resolve();
+    try {
+      await hook(receiveDeleteHighlight(highlight as unknown as Highlight, meta));
+    } catch (error) {
+      expect(deleteHighlightClient).toHaveBeenCalled();
```
Comment (PiotrKozlowski) [inline src/app/content/highlights/hooks/receiveDeleteHighlight.spec.ts:67]: there is no assumption on error type

File: src/app/content/highlights/hooks/updateHighlight.spec.ts [hunk 5]
```diff
@@ -88,7 +86,7 @@ describe('updateHighlight', () => {
     expect(updateHighlightClient).not.toHaveBeenCalled();
   });
 
-  it('doesn\'t show toast / revert update if it would result in the same highlight', async() => {
+  it('reverts update if it would result in the same highlight', async() => {
```
Comment (PiotrKozlowski) [inline src/app/content/highlights/hooks/updateHighlight.spec.ts:89]: we should probably expect an error in this test

File: src/app/content/studyGuides/hooks/printStudyGuides.spec.ts [hunk 6]
```diff
@@ -86,9 +107,25 @@ describe('printStudyGuides', () => {
     expect(calmSpy).toHaveBeenCalled();
     await Promise.resolve();
 
+    expect(dispatch).toBeCalledWith(toggleStudyGuidesSummaryLoading(false));
```
Comment (PiotrKozlowski) [inline src/app/content/studyGuides/hooks/printStudyGuides.spec.ts:110]: was this happening in the previous version?

Comment (PiotrKozlowski) [reply src/app/content/studyGuides/hooks/printStudyGuides.spec.ts:110]: probably yes but I'll check

File: src/app/content/studyGuides/hooks/printStudyGuides.spec.ts [hunk 7]
```diff
@@ -86,9 +107,25 @@ describe('printStudyGuides', () => {
...
     expect(print).toHaveBeenCalled();
   });
 
+  it('doesn\'t wait for promiseCollector.calm', async() => {
```
Comment (PiotrKozlowski) [inline src/app/content/studyGuides/hooks/printStudyGuides.spec.ts:114]: is this required for coverage for some if path?

File: src/app/utils.spec.tsx [hunk 8]
```diff
@@ -132,6 +135,30 @@ describe('actionHook', () => {
   });
 });
 
+it('handle error if it is instance of ApplicationMesssageError', async() => {
+  const hookSpy = jest.fn(async() => Promise.reject(
+    new HighlightCreateError({ destination: 'myHighlights', shouldAutoDismiss: true }
```
Comment (PiotrKozlowski) [inline src/app/utils.spec.tsx:140]: probably better to use `createApplicationError` here

**Review by PiotrKozlowski — COMMENTED** [review]
since `expect` called in `catch {}` block will throw an error only if there is an error, we should make sure that these functions are called

**Comment by jarosik10** [comment]
> since `expect` called in `catch {}` block will throw an error only if there is an error, we should make sure that these functions are called

`expect.assertions(number)` should be fine, right?

File: src/app/content/highlights/hooks/locationChange.spec.ts (hunk 2 above)
Comment (jarosik10) [reply src/app/content/highlights/hooks/locationChange.spec.ts:152]: But it doesn't `dispatch(addToast(...)` anymore. It's only about throwing the proper error.

**Comment by PiotrKozlowski** [comment]
@jarosik10 
> expect.assertions(number) should be fine, right?

yes

File: src/app/content/highlights/hooks/receiveDeleteHighlight.spec.ts (hunk 4 above)
Comment (jarosik10) [reply src/app/content/highlights/hooks/receiveDeleteHighlight.spec.ts:67]: I wanted to test `dispatch(createHighlight(...)` first and then error throwing.
```
} catch (error) {
   dispatch(createHighlight(
      {
          ...payload as unknown as NewHighlight,
          id: payload.id,
       },
       {...meta, revertingAfterFailure: true}
   ));
   
   if (error instanceof CustomApplicationError) {
      throw error;
   }
   
   throw new HighlightDeleteError({ destination });
}
```

File: src/app/content/studyGuides/hooks/printStudyGuides.spec.ts (hunk 7 above)
Comment (jarosik10) [reply src/app/content/studyGuides/hooks/printStudyGuides.spec.ts:114]: Yup. It's about `  if (waitingForPromiseCollector) {...}` else path.
Maybe the test name is misleading or smthing.

File: src/app/content/highlights/hooks/createHighlight.spec.ts (hunk 1 above)
Comment (PiotrKozlowski) [reply src/app/content/highlights/hooks/createHighlight.spec.ts:82]: its in the separate test below

File: src/app/content/highlights/hooks/locationChange.spec.ts (hunk 2 above)
Comment (PiotrKozlowski) [reply src/app/content/highlights/hooks/locationChange.spec.ts:152]: I think we should test if the hook with `receivePageFocus` will not throw `HighlightLoadError`

File: src/app/content/highlights/hooks/updateHighlight.spec.ts (hunk 5 above)
Comment (PiotrKozlowski) [reply src/app/content/highlights/hooks/updateHighlight.spec.ts:89]: nope

**Pull request merged**